expression/helper.go

```diff
@@ -136,28 +137,34 @@ func IsCurrentTimeExpr(e ast.ExprNode) bool {
 	return x.FnName.L == currentTimestampL
 }
 
-func getSystemTimestamp(ctx context.Context) (time.Time, error) {
-	value := time.Now()
-
-	if ctx == nil {
-		return value, nil
+func getSystemTimestamp(ctx context.Context) (types.Time, error) {
+	sessionVars := ctx.GetSessionVars()
+	value := types.Time{
```

Comment:
No, line 169 use it.
If "timestamp" session variable is not set, this is the default value to be returned.